cmd/byctl/ls.go

```diff
@@ -47,62 +50,105 @@ var lsCmd = &cli.Command{
 			return err
 		}
 
-		conn, path, err := cfg.ParseProfileInput(c.Args().Get(0))
-		if err != nil {
-			logger.Error("parse profile input", zap.Error(err))
-			return err
-		}
+		var chList []chan *operations.ObjectResult
+		for i := 0; i < c.Args().Len(); i++ {
+			conn, path, err := cfg.ParseProfileInput(c.Args().Get(i))
+			if err != nil {
+				logger.Error("parse profile input", zap.Error(err))
+				continue
+			}
 
-		store, err := services.NewStoragerFromString(conn)
-		if err != nil {
-			logger.Error("init storager", zap.Error(err))
-			return err
-		}
+			store, err := services.NewStoragerFromString(conn)
+			if err != nil {
+				logger.Error("init storager", zap.Error(err))
+				continue
+			}
 
-		so := operations.NewSingleOperator(store)
+			so := operations.NewSingleOperator(store)
+
+			ch, err := so.List(path)
+			chList = append(chList, ch)
+			if err != nil {
+				logger.Error("list",
+					zap.String("path", path),
+					zap.Error(err))
+				continue
+			}
+		}
 
 		// TODO: we need support more format that gnsls supports.
 		format := shortListFormat
 		if c.Bool("l") || c.String("format") == "long" {
 			format = longListFormat
 		}
 
-		ch, err := so.List(path)
-		if err != nil {
-			logger.Error("list",
-				zap.String("path", path),
-				zap.Error(err))
-			return err
-		}
-
-		isFirst := true
-		var totalNum int
-		var totalSize int64
+		// channel read status
+		mo := statsMonitor{}
+		mo.newStatMonitor(len(chList))
 
-		for v := range ch {
-			if v.Error != nil {
-				logger.Error("read next result", zap.Error(v.Error))
-				return v.Error
-			}
+		// cache result
+		var resultList = make([][]*operations.ObjectResult, c.Args().Len())
 
-			oa := parseObject(v.Object)
-			fmt.Print(oa.Format(format, isFirst))
+		// search finished first, print first
+		cases := make([]reflect.SelectCase, len(chList))
+		for i, ch := range chList {
+			cases[i] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(ch)}
+		}
 
-			// Update isFirst
-			if isFirst {
-				isFirst = false
+		isFirstSrc := true
+		for {
+			index, value, _ := reflect.Select(cases)
```

Comment:
I think we have introduced too much complex logic (and even reflect) to support the calculation of total size and count.

How about support a callback in `so.ListWithCallback(path, fn func(o *types.Object))`?

And maybe we can refactor the whole `cmd` package in the futher.